Add django.contrib.postgres.forms*

asfaltboy opened the pull request:
generated via stubgen and cleaned up a bit

asfaltboy committed:
Add django.contrib.postgres.forms

generated via stubgen and cleaned up a bit

mkurnikov commented on django-stubs/contrib/postgres/forms/array.pyi:
```diff
@@ -0,0 +1,41 @@
+from django import forms as forms
+
+class SimpleArrayField(forms.CharField):
+    default_error_messages: Any = ...
```
If attribute is defined on the base class, no need to define it on every field.

asfaltboy replied:
addressed in 55abc1e77464908da2d53dd30f27b704c16d6d3d

mkurnikov reviewed (changes requested):
Overall, I'd prefer having as little `Any` throughout the stubs as possible. Better have no method defined than `Any`. So, please, add types instead. 
Thanks!

asfaltboy committed:
Remove some method types, specify others' types

asfaltboy committed:
Respecify some inherited attribute types

For some reason using the inherited ones fails this test case:

https://travis-ci.com/typeddjango/django-stubs/jobs/273220761

I'm not really sure why this is :/

mkurnikov commented on django-stubs/contrib/postgres/forms/ranges.pyi:
```diff
@@ -1,17 +1,24 @@
-class DateTimeRangeField(BaseRangeField): ...
-class DateRangeField(BaseRangeField): ...
+class IntegerRangeField(BaseRangeField):
+    base_field: Union[Type[forms.Field], forms.Field] = ...
```
`base_field` defined on `forms.Field`, I think. If it may be both `Type[Field]` and `Field`, better fix it there, and remove all this duplication.

asfaltboy replied:
addressed in 67d664d

mkurnikov commented on django-stubs/contrib/postgres/forms/jsonb.pyi:
```diff
@@ -1,8 +1,9 @@
 class JSONString(str): ...
 
 class JSONField(forms.CharField):
+    widget: Union[forms.Widget, Type[forms.Widget]] = ...
```
Same as in the comment about `base_field`

asfaltboy replied:
Django is weird ... the field classes expect `widget` to define a static attribute specifying the widget class, but then instantiate and set the `self.widget` to the instance in the constuctor 🤦‍♂ ... so if I pull the above line to `forms.Field` I start seeing errors like this:
```
django-sources/tests/postgres_tests/test_ranges.py:723: error: Argument 1 to "render" of "Widget" has incompatible type "str"; expected "Widget"
django-sources/tests/postgres_tests/test_ranges.py:723: error: Too few arguments for "render" of "Widget"
django-sources/tests/postgres_tests/test_ranges.py:727: error: Argument 1 to "render" of "Widget" has incompatible type "str"; expected "Widget"
django-sources/tests/postgres_tests/test_ranges.py:727: error: Argument 2 to "render" of "Widget" has incompatible type "None"; expected "str"
```

I could only resolve this by pulling defining this in child class:

```
class BaseRangeField(forms.MultiValueField):
    widget: Widget
```

In that case, I guess it's better to keep the `Union` on the parent, and add the `Argument 1 to "render" of "Widget" has incompatible type "str"` error under: `enabled_test_modules.py`?

asfaltboy committed:
Replace duplicate definitions with error ignores
